source/src/plugins/KolmogorovTest.cc

```diff
@@ -0,0 +1,151 @@
+
+    protected:
+      std::string m_options;
+
+    };
+
+    typedef KolmogorovTest::Factory KolmogorovTestFactory;
+
+    //-------------------------------------------------------------------------------------------------
+    //-------------------------------------------------------------------------------------------------
+
+    inline QTestPtr KolmogorovTest::Factory::createQualityTest(const std::string &qname) const {
+      return std::shared_ptr<QTest>(new KolmogorovTest(qname));
+    }
+
+    //-------------------------------------------------------------------------------------------------
+    //-------------------------------------------------------------------------------------------------
+
+    KolmogorovTest::KolmogorovTest(const std::string &qname)
+        : QualityTest("Kolmogorov", qname),
+	  m_options()
+{
+      m_description = "Performs the Kolmogorov-Smirnov test on a monitor element and a reference, outputting the p-value. In general this should only "
+	              "be used for TGraphs. While this test can take TH1s, the Kolmogorov test is intended for use on unbinned data, not histograms. "
+	              "See https://root.cern.ch/doc/master/classTH1.html#aeadcf087afe6ba203bcde124cfabbee4 for more information.";
+    }
+
+    //-------------------------------------------------------------------------------------------------
+
+    StatusCode KolmogorovTest::readSettings(const TiXmlHandle xmlHandle) {
+      RETURN_RESULT_IF_AND_IF(STATUS_CODE_SUCCESS, STATUS_CODE_NOT_FOUND, !=, XmlHelper::readParameter(xmlHandle, "Options", m_options));
+
+      return STATUS_CODE_SUCCESS;
+    }
+
+    void KolmogorovTest::userRun(MonitorElement* pMonitorElement, QualityTestReport &report) {
+      const bool hasObject = (pMonitorElement->object() != nullptr);
+      const bool hasReference = (pMonitorElement->reference() != nullptr);
+      const bool isHistogram = pMonitorElement->object()->InheritsFrom("TH1");
+      const bool isGraph = pMonitorElement->object()->InheritsFrom("TGraph");
+
+      if (!hasObject) {
+        report.m_message = "No object attached to monitor element";
+        report.m_quality = 0.f;
+        throw StatusCodeException(STATUS_CODE_INVALID_PTR);
+      }
+
+      if (!hasReference) {
+        report.m_message = "No reference attached to monitor element";
+        report.m_quality = 0.f;
+        throw StatusCodeException(STATUS_CODE_INVALID_PTR);
+      }
+
+      if (isGraph) {
+	TGraph* pGraph = pMonitorElement->objectTo<TGraph>();
+	TGraph* pReferenceGraph = pMonitorElement->referenceTo<TGraph>();
+
+	Option_t* pOptions = &m_options[0u];
+
+	int sizeGraph = pGraph->GetN();
+	int sizeRef = pReferenceGraph->GetN()  ;
+	double* pArrayGraph = pGraph->GetY();
+	double* pArrayRef = pReferenceGraph->GetY();
+
+	std::sort(&pArrayGraph[0], &pArrayGraph[sizeGraph]);
+	std::sort(&pArrayRef[0], &pArrayRef[sizeRef]);
+       
+	report.m_message = ("Options used: " + m_options);
```

Comment:
Use the `report.m_extraInfos` to add custom output to the report.
See example here : https://github.com/DQM4HEP/dqm4hep-core/blob/master/source/src/plugins/FitParamInRangeTest.cc#L410